runtime/lua/vim/uri.lua

```diff
@@ -110,7 +110,13 @@ local function uri_to_fname(uri)
   -- TODO improve this.
   if is_windows_file_uri(uri) then
     uri = uri:gsub('^file:/+', '')
-    uri = uri:gsub('/', '\\')
+    if vim.fn.has('wsl') == 1 then
+      uri = uri:gsub('^([a-zA-Z]):', function(match)
+        return '/mnt/'.. match:lower()
```

Comment:
Isn't this assuming that the uri is an absolute path?

I also don't even think you should be running `clangd.exe` from inside WSL. And if you aren't already, then you'll probably end up facing issues with `clangd` itself, trying to decide its sysroot.

There can generally be more support for WSL, but I honestly don't think this is robust enough.